gluten-data/src/main/scala/io/glutenproject/vectorized/ColumnarBatchSerializer.scala

```diff
@@ -70,56 +69,66 @@ private class ColumnarBatchSerializerInstance(
   extends SerializerInstance
   with Logging {
 
+  private lazy val shuffleReaderHandle = {
```

Comment:
Not directly related to this PR. It seems [CelebornColumnarBatchSerializer.scala](https://github.com/oap-project/gluten/pull/2802/files#diff-89f2f350ebb871afee74677e5c52b05c85795d3f07080d0edd1895b2bf703f80) quite similar to this class. Can we reuse this?

Reply:
It makes a lot of sense to me.

The consolidation of the two serializers was in our original plan https://github.com/oap-project/gluten/issues/1205

> In Java, consolidate GlutenColumnarBatchSerializer and CelebornColumnarBatchSerializer

But somehow we didn't make the change yet. We should do it asap since the code brings extra maintenance burden to developers.

Reply:
#1205 was closed. How about creating another issues tracking this? Thoughts?